### PR TITLE
Added complex data support for NxSpectrum

### LINKED ISSUE #310: [Nexus] Add support for complex data
By complex data, I mean numbers that have a real and an imaginary part :wink:

### PR DESCRIPTION
**Pull request by loichuder**
Closes #310

### CODE REVIEW

File: cypress/specs/app.spec.ts [hunk 1]
```diff
@@ -66,6 +82,11 @@ describe('App', () => {
     );
     cy.findByRole('tab', { name: 'Heatmap' }).should('exist');
     cy.findByRole('figure', { name: /twoD_cplx/ }).should('exist');
+
+    if (!!Cypress.env('TAKE_SNAPSHOTS')) {
+      cy.wait(SNAPSHOT_DELAY);
+      cy.matchImageSnapshot('heatmap_2D_complex');
+    }
```
Comment (loichuder) [inline cypress/specs/app.spec.ts:89]: I added e2e tests for complex line and heatmap.

Do you think it is worth adding them for NeXus as well ?

File: src/h5web/vis-packs/nexus/models.ts [hunk 2]
```diff
@@ -26,17 +26,19 @@ export enum NxInterpretation {
   Image = 'image',
 }
 
-export interface NxData {
-  signalDataset: Dataset<ArrayShape, NumericType | ComplexType>;
+export interface NxData<
+  T extends NumericType | ComplexType = NumericType | ComplexType
+> {
+  signalDataset: Dataset<ArrayShape, T>;
   errorsDataset?: NumArrayDataset;
```
Comment (loichuder) [inline src/h5web/vis-packs/nexus/models.ts:33]: In the end, I kept `NumArrayDataset` for `errorsDatasets` and `auxDatasets`. I had issues with the conditional type :confused:

File: src/h5web/vis-packs/nexus/visualizations.ts [hunk 3]
```diff
@@ -35,11 +47,11 @@ export const NEXUS_VIS: Record<NexusVis, VisDef> = {
     ConfigProvider: HeatmapConfigProvider,
   },
 
-  [NexusVis.NxComplex]: {
+  [NexusVis.NxComplexImage]: {
     name: NexusVis.NxImage,
     Icon: FiMap,
     Toolbar: ComplexToolbar,
-    Container: NxComplexContainer,
+    Container: NxComplexImageContainer,
```
Comment (loichuder) [inline src/h5web/vis-packs/nexus/visualizations.ts:54]: Renamed `NxComplex` into `NxComplexImage` in accordance with `NxComplexSpectrum`

File: cypress/specs/app.spec.ts (hunk 1 above)
Comment (axelboc) [reply cypress/specs/app.spec.ts:89]: Neh, not worth it, I reckon. Did you add feature tests, though? Those are more important.

**Review by axelboc — APPROVED** [review]

File: cypress/specs/app.spec.ts (hunk 1 above)
Comment (loichuder) [reply cypress/specs/app.spec.ts:89]: This: https://github.com/silx-kit/h5web/pull/667/files#diff-217560973575044a39d13ed85c227f1cd30430e8db25666e0a013bd8399ab72f you mean ?

**Pull request merged**